### PR TITLE
Issue2886 fan coil model

### PR DESCRIPTION
**Pull request by karthikeyad-pnnl**
@SenHuang19 : I am submitting this PR to merge in the files for the fan coil unit system model. Please review this PR after merging in the changes in #1.

### CODE REVIEW

File: Buildings/Fluid/ZoneEquipment/FCUControls/Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.mo [hunk 1]
```diff
@@ -0,0 +1,94 @@
+within Buildings.Fluid.ZoneEquipment.FCUControls;
+block Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate
+  Controls.OBC.CDL.Interfaces.RealInput TZon "Measured zone temperature"
+    annotation (Placement(transformation(extent={{-140,40},{-100,80}}),
+        iconTransformation(extent={{-140,20},{-100,60}})));
+  Controls.OBC.CDL.Interfaces.RealInput TCooSet "Measured zone temperature"
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/FCUControls/Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.mo]: Correct the comment

File: Buildings/Fluid/ZoneEquipment/FCUControls/Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.mo [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    annotation (Placement(transformation(extent={{-140,0},{-100,40}}),
+        iconTransformation(extent={{-140,-20},{-100,20}})));
+  Controls.OBC.CDL.Interfaces.RealInput THeaSet "Measured zone temperature"
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/FCUControls/Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.mo]: Correct the comment

File: Buildings/Fluid/ZoneEquipment/FCUControls/Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.mo [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    annotation (Placement(transformation(extent={{0,10},{20,30}})));
+  Controls.OBC.CDL.Logical.Or or2
+    annotation (Placement(transformation(extent={{60,-70},{80,-50}})));
+  Controls.OBC.CDL.Continuous.PID conPID(reverseActing=false)
+    annotation (Placement(transformation(extent={{0,-20},{20,0}})));
+  Controls.OBC.CDL.Continuous.PID conPID1(reverseActing=false)
+    annotation (Placement(transformation(extent={{0,-50},{20,-30}})));
+  Controls.OBC.CDL.Continuous.Sources.Constant con(k=0)
+    annotation (Placement(transformation(extent={{-80,-90},{-60,-70}})));
+  Controls.OBC.CDL.Continuous.Add add3
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/FCUControls/Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.mo]: Add comments

File: Buildings/Fluid/ZoneEquipment/Data/FCUSizing.mo [hunk 4]
```diff
@@ -0,0 +1,84 @@
+within Buildings.Fluid.ZoneEquipment.Data;
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/Data/FCUSizing.mo:1]: Maybe we should create a sub-package called FanCoil and put the Data package under it

File: Buildings/Fluid/ZoneEquipment/Data/FCUSizing.mo [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+  parameter Modelica.Units.SI.SpecificHeatCapacity cp_air = 1000;
+
+  parameter Modelica.Units.SI.ThermalConductance UACooCoiExt_nominal = cp_air * qCooCoi_nominal/delH_lmd;
+
+  parameter Modelica.Units.SI.ThermalConductance UACooCoiTot_nominal = 1/(1/UACooCoiExt_nominal + 1/(3.3*UACooCoiExt_nominal));
+
+  annotation (Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
+        coordinateSystem(preserveAspectRatio=false)));
+end FCUSizing;
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/Data/FCUSizing.mo:88]: Add documentation and revision history

File: Buildings/Fluid/ZoneEquipment/Data/package.mo [hunk 6]
```diff
@@ -0,0 +1,6 @@
+within Buildings.Fluid.ZoneEquipment;
+package Data
+
+  extends Modelica.Icons.MaterialPropertiesPackage;
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/Data/package.mo:4]: Add documentation

File: Buildings/Fluid/ZoneEquipment/FCUControls/Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.mo [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+          {28,-14}}, color={0,0,127}));
+  connect(conPID1.y, add3.u2) annotation (Line(points={{22,-40},{26,-40},{26,
+          -26},{28,-26}}, color={0,0,127}));
+  annotation (Icon(coordinateSystem(preserveAspectRatio=false), graphics={
+          Rectangle(
+          extent={{-100,100},{100,-100}},
+          lineColor={0,0,0},
+          fillColor={255,255,255},
+          fillPattern=FillPattern.Solid)}), Diagram(coordinateSystem(
+          preserveAspectRatio=false)));
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/FCUControls/Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.mo]: Add documentations and revision history

File: Buildings/Fluid/ZoneEquipment/FCUControls/Validation.mo [hunk 8]
```diff
@@ -0,0 +1,32 @@
+within Buildings.Fluid.ZoneEquipment.FCUControls;
+package Validation
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/FCUControls/Validation.mo:2]: Add documentation and revision history

File: Buildings/Fluid/ZoneEquipment/FCUControls/Validation.mo [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      annotation (Line(points={{-38,20},{-30,20},{-30,0},{-10,0}}, color={0,0,127}));
+    connect(con1.y, controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.THeaSet)
+      annotation (Line(points={{-38,-20},{-30,-20},{-30,-4},{-10,-4}}, color={0,0,
+            127}));
+    connect(sin.y, controller_MultiSpeedCyclingFan_ConstantWaterFlowrate.TZon)
+      annotation (Line(points={{-58,50},{-20,50},{-20,4},{-10,4}}, color={0,0,127}));
+    annotation (Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
+          coordinateSystem(preserveAspectRatio=false)),
+      experiment(StopTime=60, __Dymola_Algorithm="Dassl"));
+  end Controller_MultiSpeedCyclingFan_ConstantWaterFlowrate;
```
Comment (SenHuang19) [inline Buildings/Fluid/ZoneEquipment/FCUControls/Validation.mo:40]: Add a script

**Comment by karthikeyad-pnnl** [comment]
@SenHuang19 I have updated the package structure for the fan coil unit. Please review it and let me know if you run into any issues while running the simulations. Thanks!

**Pull request merged**